plcc/src/proteinstructure/Molecule.java

```diff
@@ -482,12 +494,118 @@ private Integer centroidDistTo(Molecule m) {
+//            }
+//            catch(Exception e) {
+//                if( ! Settings.getBoolean("plcc_B_no_parse_warn")) {
+//                    DP.getInstance().w("Could not determine distance between DSSP residues " + this.getDsspNum() + " and " + r.getDsspNum() + ", assuming out of contact distance.");
+//                }
+//                return(false);
+//            }
+//            Integer atomRadius;
+//            if(this.isLigand() || r.isLigand()) {
+//                atomRadius = Settings.getInteger("plcc_I_lig_atom_radius");
+//            }
+//            else if (this.isAA() || r.isAA()) {
+//                atomRadius = Settings.getInteger("plcc_I_aa_atom_radius");
+//            }
+//            else {
+//                atomRadius = Settings.getInteger("plcc_I_rna_atom_radius");
+//            }
+//
+//            Integer justToBeSure = 4;   // Setting this to 0 shouldn't change the number of contacts found (but all harm it could do is to increase the runtime a tiny bit). Verified: has no influence. Should be removed in future release.
+//            Integer maxDistForContact = this.getCenterSphereRadius() + r.getCenterSphereRadius() +  (atomRadius * 2) + justToBeSure;
+//
+//            //System.out.println("    Center sphere radius for PDB residue " + this.getPdbResNum() + " = " + this.getCenterSphereRadius() + ", for " + r.getPdbResNum() + " = " + r.getCenterSphereRadius() + ", atom radius is " + atomRadius + ".");
+//            //System.out.println("    DSSP Res distance " + this.getDsspResNum() + "/" + r.getDsspResNum() + " is " + dist + " (no contacts possible above distance " + maxDistForContact + ").");
+//
+//            if(dist > (maxDistForContact)) {
+//                return(false);
+//            }
+//            else {
+//                return(true);
+//            }
+//        }
+//        else {
+//            DP.getInstance().w("Tried to get distance between Molecules. Due to ugly code this is not possible atm.");
+//        }
+//
+//        // if one of them is Molecule or RNA
+//        return false;
+//    }
 
     // ugly abstract methods
-    abstract public Integer getType();
     abstract public Boolean isAA();
     abstract public Boolean isLigand();
+    
+
+    /**
+     * Returns the type-specific atom radius of the component (type = Residue, Ligand, Other, RNA).
+     */
+    public Integer getAtomRadius() {
+        //start with instanceOf Molecule??
+        Integer atomRadius = null;
+        
+        switch(type) {
+            case 0:
+                atomRadius = Settings.getInteger("plcc_I_aa_atom_radius");
+            case 1:
+                atomRadius = Settings.getInteger("plcc_I_lig_atom_radius");       // no value for "Other" so far --> Default is AA
+            case 2:
+                atomRadius = Settings.getInteger("plcc_I_aa_atom_radius");
+            case 3:
+                atomRadius = Settings.getInteger("plcc_I_rna_atom_radius");
```

Comment:
I am curious, have you tested it with ligands? Because of the fall through it should not work properly, in my opinion. Tutorial on fall throughs: https://www.tutorialspoint.com/Java-fall-through-switch-statements .